lib/matplotlib/artist.py

```diff
@@ -1303,10 +1304,22 @@ def format_cursor_data(self, data):
                 return "[]"
             normed = self.norm(data)
             if np.isfinite(normed):
-                # Midpoints of neighboring color intervals.
-                neighbors = self.norm.inverse(
-                    (int(self.norm(data) * n) + np.array([0, 1])) / n)
-                delta = abs(neighbors - data).max()
+                if isinstance(self.norm, BoundaryNorm):
+                    # not an invertible normalization mapping
+                    cur_idx = np.argmin(np.abs(self.norm.boundaries - data))
+                    neigh_idx1 = max(0, cur_idx - 1)
```

Comment:
I think you could write this as 

```python
delta = np.diff(self.norm.boundaries[np.clip(cur_indx-1, 0):cur_index+2]).max()
```

which I think is a bit clearer?

Reply:
Slicing makes it cleaner, true. But I think `np.clip` needs both min and max bounds.

Reply:
oh, fair.  The dangers of typing code off the top of ones head into a browser ;)